File: Safari-611.1.10/LayoutTests/http/tests/navigation/post-redirect-get-reload.php
<?php
if($_SERVER['REQUEST_METHOD'] == "POST") {
    header("Location: post-redirect-get-reload.php", true, 303);
    exit;
}
?>
<body>
1. Submit a form<br>
1a. The form redirects to a get.<br>
2. Reload<br><br>
The reload should not trigger a form resubmission warning.

<form name="form" action="post-redirect-get-reload.php" method="post"><input type="submit"></input></form>
<script>

if (window.testRunner) {
    testRunner.dumpAsText();
    testRunner.waitUntilDone();

    if (window.sessionStorage["prgl-state"] == null) {
        window.sessionStorage["prgl-state"] = 'submitted';
        document.form.submit();
    } else {
        window.sessionStorage.clear();
        testRunner.waitForPolicyDelegate();
        window.internals.forceReload(false);
    }
}
</script>

File: LayoutTests/http/tests/navigation/post-redirect-get-reload.py
#!/usr/bin/env python3

import os
import sys

method = os.environ.get('REQUEST_METHOD', '')

sys.stdout.write('Content-Type: text/html\r\n')

if method == 'POST':
    sys.stdout.write(
        'Location: post-redirect-get-reload.py\r\n'
        'status: 303\r\n\r\n'
    )
    sys.exit(0)

sys.stdout.write(
    '\r\n'
    '<body>\n'
    '1. Submit a form<br>\n'
    '1a. The form redirects to a get.<br>\n'
    '2. Reload<br><br>\n'
    'The reload should not trigger a form resubmission warning.\n'
    '\n'
    '<form name="form" action="post-redirect-get-reload.py" method="post"><input type="submit"></input></form>\n'
    '<script>\n'
    '\n'
    'if (window.testRunner) {\n'
    '    testRunner.dumpAsText();\n'
    '    testRunner.waitUntilDone();\n'
    '\n'
    '    if (window.sessionStorage["prgl-state"] == null) {\n'
    '        window.sessionStorage["prgl-state"] = \'submitted\';\n'
    '        document.form.submit();\n'
    '    } else {\n'
    '        window.sessionStorage.clear();\n'
    '        testRunner.waitForPolicyDelegate();\n'
    '        window.internals.forceReload(false);\n'
    '    }\n'
    '}\n'
    '</script>\n'
)